shell/browser/electron_browser_main_parts.cc

```diff
@@ -375,8 +378,23 @@ void ElectronBrowserMainParts::PostDestroyThreads() {
   fake_browser_process_->PostDestroyThreads();
 }
 
+#if defined(OS_LINUX)
+// static
+const char* ElectronBrowserMainParts::GetGDKBackend() {
+  return gdk_backend;
+}
+#endif
+
 void ElectronBrowserMainParts::ToolkitInitialized() {
 #if defined(OS_LINUX)
+  // This is set by Chromium here:
+  // https://chromium-review.googlesource.com/c/chromium/src/+/2586184
+  // and can detrimentally affect external app behaviors, so we want to
+  // check if the user has set it so we can use it later.
+  std::string backend;
+  if (base::Environment::Create()->GetVar("GDK_BACKEND", &backend))
+    gdk_backend = backend.c_str();
```

Comment:
`backend` is a local variable and once `ToolkitInitialized ` finishes the `const char*` will point to freed memory. A common pattern for this kind of cache is:

```c++
base::Optional<std::string>& ElectronBrowserMainParts::GetGDKBackend() {
  static base::Optional<std::string> gdk_backend;
  return gdk_backend;
}

void ElectronBrowserMainParts::ToolkitInitialized() {
  ...
  GetGDKBackend().reset(backend);
  ...
}
```